### PR TITLE
docs: spelling, grammar, and consistency pass over website

### PR DESCRIPTION
**Pull request by jvanstraten**
- Spelling, grammar, and punctuation pass using languagetool and manual scanning through all pages of the website.
 - Replaced all instances of () parentheses for type parameters with <>, in accordance with https://substrait.io/types/type_parsing/ and the ANTLR grammar.
 - Used backticks to distinguish types etc. from normal words in more places (though not everywhere, because it messes up table formatting apparently).
 - Only fixed semantic things that were very obviously wrong, like i32 being defined as a 32-byte integer instead of 32-bit.

### CODE REVIEW

**Comment by jacques-n** [comment]
Ping here on the rebase

**Comment by cpcloud** [comment]
I handled the rebase. Merging shortly.

File: site/docs/expressions/scalar_functions.md [hunk 1]
```diff
@@ -128,8 +128,14 @@ For reference, here are are some common output type derivations and how they can
 | Operation                                                    | Definition                                                   |
 | ------------------------------------------------------------ | ------------------------------------------------------------ |
 | Add item to list                                             | `add(<List<T>, T>) => List<T>`                               |
+<<<<<<< Updated upstream
```
[the remark this answers is not in the document]
Comment (cpcloud) [reply site/docs/expressions/scalar_functions.md]: Fixed.

**Comment by cpcloud** [comment]
Thanks @jvanstraten!

**Pull request merged**